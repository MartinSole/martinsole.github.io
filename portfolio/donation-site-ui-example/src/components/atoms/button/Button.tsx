interface Props {
  children: string
  buttonType?: "submit" | "reset" | "button" | undefined
}

const Button = ({buttonType = undefined, children}: Props) => {
  return (
    <button className='buttonBase' type={buttonType}>{children}</button>
  )
}

export default Button