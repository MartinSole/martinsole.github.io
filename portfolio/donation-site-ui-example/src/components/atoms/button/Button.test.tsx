import '@testing-library/jest-dom'
import { render } from "@testing-library/react"
import Button from "./Button"

test("Renders the Button component", () => {
   const {container, findByText} = render(<Button children="test" />)
    expect(container).toMatchSnapshot()
    expect(findByText('test')).toBeTruthy
})
