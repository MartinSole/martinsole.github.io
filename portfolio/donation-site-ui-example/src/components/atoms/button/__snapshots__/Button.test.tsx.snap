// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Button component 1`] = `
<div>
  <button
    class="buttonBase"
  >
    test
  </button>
</div>
`;
